### DISCUSSION TITLE
Can't compile hello world in pvsneslib

### DISCUSSION THREAD

**Discussion by bkumanchik** [alekmaul/pvsneslib#262] [General]
Can't compile hello world in pvsneslib, I get:

make: *** No targets specified and no makefile found.  Stop.

**Comment by alekmaul** [comment]
I think it is better to use our discord for such issue, but:
- are you sure you followed https://github.com/alekmaul/pvsneslib/wiki/Installation?
- how do you tried  to compile, inside an IDE, in a CLI?

**Reply by bkumanchik** [reply]
yes, where do I actually place these lines, how do I find the location? the installation tutorial is great but I can't figure out where to find the location to put these two lines and do I just add them or replace some existing lines, I got it working once before but can't get it working now.

------------------------
If you use the integrated terminal in vscode, you will have to update the user settings by adding this block :

"terminal.integrated.profiles.windows": { "MSYS2 UCRT": { "path": "cmd.exe", "args": [ "/c", "C:\\msys64\\msys2_shell.cmd -defterm -here -no-start -ucrt64" ] } },

and the workspace settings with this one :

"terminal.integrated.defaultProfile.windows": "MSYS2 UCRT"
------------------------

**Reply by alekmaul** [reply]
hummm, why the current vscode integration is not working for you? do you really nee dto update it?

**Comment by bkumanchik** [comment]
I've made a video of my complete setup process:

https://www.dropbox.com/scl/fi/oixncs50rczsunc2xtjgy/PVSnes_setup.mp4?rlkey=zdshn89g5a8od78kx990zdbm2&dl=0

Maybe you can see what I'm doing wrong.

Thanks,

Brian

**Reply by alekmaul** [reply]
sorry, i do not want to take some dropbox links. put it on youtube if you can ;)

**Reply by bkumanchik** [reply]
OK, here it is:

https://www.youtube.com/watch?v=rCW9r8GaCFo

**Reply by alekmaul** [reply]
Thanks a lot for the video, that's really great to see it in action. I'm also using vscode and I do not have the problem.
Here are some tests to do:
- could you please try to remove the settings.json configuration of msys2, i can't remember why we put it here, but perhaps your problem is from that configuration. Try again with the default powershell and tell me if it is better.
- Also, if you try with Ctrl+Shift+B and Clean then Ctrl+Shirt+B and Build, does it work? Normally, it launches a terminal command with "make clean" then "make"

**Reply by bkumanchik** [reply]
Are you saying that I don't need msys2 at all?

**Reply by alekmaul** [reply]
no, just that i don't think we need to install msys2 inside vscode (the last chapter in the wiki with vscode tutorial)

**Comment by bkumanchik** [comment]
OK,  [make clean]  with powershell gives me:

**Makefile:5: /c/snesdev/devkitsnes/snes_rules: No such file or directory
make: *** No rule to make target '/c/snesdev/devkitsnes/snes_rules'.  Stop.
PS C:\snesdev\snes-examples\hello_world>** 

[make] with powershell gives me: (same thing)

**Makefile:5: /c/snesdev/devkitsnes/snes_rules: No such file or directory
make: *** No rule to make target '/c/snesdev/devkitsnes/snes_rules'.  Stop.
PS C:\snesdev\snes-examples\hello_world>** 

CTRL+SHIFT+B then PVSlib - Clean gives me:

 **  Executing task: make clean 

Makefile:5: /c/snesdev/devkitsnes/snes_rules: No such file or directory
make: *** No rule to make target '/c/snesdev/devkitsnes/snes_rules'.  Stop.

 *  The terminal process "C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -Command make clean" terminated with exit code: 1. 
 *  Press any key to close the terminal.**


CTRL+SHIFT+B then PSVLib - Build gives me:

 **  Executing task: make all 

Makefile:5: /c/snesdev/devkitsnes/snes_rules: No such file or directory
make: *** No rule to make target '/c/snesdev/devkitsnes/snes_rules'.  Stop.

 *  The terminal process "C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -Command make all" terminated with exit code: 1. 
 *  Press any key to close the terminal.**

**Reply by alekmaul** [reply]
That's really weird before i just test:
- 1) I put the .vscode folder inside the hello_world example
- 2) I open vscode and choose "File/Open folrder..." then go inside hello_world folder
- 3) I click on makefile for example (but I don't think it is mandatory)
- 4) CTRL+SHIFT+B then Clean
![image](https://github.com/alekmaul/pvsneslib/assets/2528347/d4c44835-c36f-4851-adca-3dfdf9018895)
- 5) I have this in powershell terminal
![image](https://github.com/alekmaul/pvsneslib/assets/2528347/3a5e1a03-6c5f-4dc3-9978-2db59da43c05)

If I open a terminal in vscode and try a clean, it works too
![image](https://github.com/alekmaul/pvsneslib/assets/2528347/14c0bad6-bdc4-43a0-b71b-b880aaf3bcfc)

What do you have when you open a terminal INSIDE vscode (with "TErminal/New terminal"? the hello_world directory?

I'm using W11, with no MSYS2 because you only need MSYS2 if you want to compile pvsneslib or if you don't have a make environment (i'm using a make environment from my NDS developement).

**Comment by bkumanchik** [comment]
I'll try this and let you know ASAP, thanks for helping me sort this out, if I can get it to work again, I could make a working video that you could link to.

Sent from Yahoo Mail on Android 
 
  On Sun, Feb 11, 2024 at 12:40 AM, ***@***.***> wrote:   


That's really weird before i just test:
   
   -       
      - I put the .vscode folder inside the hello_wolrd example

   -       
      - I open vscode and choose "File/Open folrder..." then go inside hello_world folder

   -       
      - I click on makefile for example (but I don't think it is mandatory)

   -       
      - CTRL+SHIFT+B then Clean      
image.png (view on web)

   -       
      - I have this in powershell terminal      
image.png (view on web)


If I open a terminal in vscode and try a clean, it works too
image.png (view on web)

What do you have when you open a terminal INSIDE vscode? the hello_world directory?

I'm using W11, with no MSYS2 because you only need MSYS2 if you want to compile pvsneslib or if you don't have a make environment (i'm using a make environment from my NDS developement).

**Comment by bkumanchik** [comment]
I get this when I try Clean:
 *  Executing task: make clean 
Makefile:5: /c/snesdev/devkitsnes/snes_rules: No such file or directorymake: *** No rule to make target '/c/snesdev/devkitsnes/snes_rules'.  Stop.
 *  The terminal process "C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -Command make clean" terminated with exit code: 1.  *  Press any key to close the terminal. 


and this when I try Build: (same thing)
 *  Executing task: make all 
Makefile:5: /c/snesdev/devkitsnes/snes_rules: No such file or directorymake: *** No rule to make target '/c/snesdev/devkitsnes/snes_rules'.  Stop.
 *  The terminal process "C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -Command make all" terminated with exit code: 1.  *  Press any key to close the terminal. 
 Also when I open a folder for the first time I always get this: (and I click Yes)



Brian
    On Sunday, February 11, 2024 at 12:40:27 AM EST, alekmaul ***@***.***> wrote:  
 
 


That's really weird before i just test:
   
   -       
      - I put the .vscode folder inside the hello_wolrd example

   -       
      - I open vscode and choose "File/Open folrder..." then go inside hello_world folder

   -       
      - I click on makefile for example (but I don't think it is mandatory)

   -       
      - CTRL+SHIFT+B then Clean      
image.png (view on web)

   -       
      - I have this in powershell terminal      
image.png (view on web)


If I open a terminal in vscode and try a clean, it works too
image.png (view on web)

What do you have when you open a terminal INSIDE vscode? the hello_world directory?

I'm using W11, with no MSYS2 because you only need MSYS2 if you want to compile pvsneslib or if you don't have a make environment (i'm using a make environment from my NDS developement).

**Comment by alekmaul** [comment]
If I look carefuly in your video, it seems that you use a snapshot and not the last release?
why not using a release for a first try?
if you are using a snapshot, you will have to compile pvsneslib to have the correct .obj files before compiling hello_world example.
also, your PVSNESLIBHOME declaration is a little blur on the video, what is the directory you declared?

**Reply by bkumanchik** [reply]
the last release only has 3 folders devkitnes, pvsneslib and snes-examples. it seems to be missing some folders.

Brian

**Reply by bkumanchik** [reply]
like the .vscode folder for example and some other files, etc.

**Reply by bkumanchik** [reply]
is last release 4.1.0?

**Comment by bkumanchik** [comment]
You mean the Environment Variable? if so it's:  [PVSNESLIB_HOME]  (with underscore - not like you have it typed above) and  [/c/snesdev]  for path
    On Sunday, February 11, 2024 at 09:15:55 AM EST, alekmaul ***@***.***> wrote:  
 
 


If I look carefuly in your video, it seems that you use a snapshot and not the last release?
why not using a release for a first try?
if you are using a snapshot, you will have to compile pvsneslib to have the correct .obj files before compiling hello_world example.
also, your PVSNESLIBHOME declaration is a little blur on the video, what is the directory you declared?

**Reply by alekmaul** [reply]
what about the release version? did you tested it instead of the snapshot one?

**Reply by bkumanchik** [reply]
yes, here's a new simplified video with a fresh install (no msys2) maybe you can see what's happening here:

https://www.youtube.com/watch?v=IKVicb-6JEI

**Reply by alekmaul** [reply]
ok, this video is really better and a good start to help you.
What do you have in /c/snesdev/devkitsnes/ ? is snes_rules inside it?
Regarding vscode, do you have C/C++ extension & extension pack, do you have makefile tools ?
I have this on my side:
![image](https://github.com/alekmaul/pvsneslib/assets/2528347/f4e3d23f-30ff-4d8a-8fe2-484367d78c4f)
I did the same thing you did to ope a terminal window (right button on makefile) and it works, it is really weird :(

when you type make -v in the vscode terminal windows, what is the result? I have this:
![image](https://github.com/alekmaul/pvsneslib/assets/2528347/9033d4e7-de30-4430-9a94-9ea1a08c2adc)

Another point, we are not compatible with git bash for compilation, i currently don't know why.

**Reply by bkumanchik** [reply]
I DO have snes_rules inside /c/snesdev/devkitsnes, but I've only installed the first of the above C++ extensions, I'll try the other two and see if that makes a difference.

**Reply by bkumanchik** [reply]
I have these installed now:
<img width="239" alt="Screenshot 2024-02-13 085206" src="https://github.com/alekmaul/pvsneslib/assets/25464489/2074b207-6558-4dd4-a27b-7ca13e639d5b">


My Environment Variables setup:
<img width="232" alt="Screenshot 2024-02-13 085414" src="https://github.com/alekmaul/pvsneslib/assets/25464489/c9ee70c9-4342-48cf-bd30-009af31d69f1">


My directory structure:
<img width="594" alt="Screenshot 2024-02-13 085514" src="https://github.com/alekmaul/pvsneslib/assets/25464489/9c2e4566-3c0f-4aec-ae1c-4a6e301ee3cf">


My project folder:
<img width="840" alt="Screenshot 2024-02-13 085747" src="https://github.com/alekmaul/pvsneslib/assets/25464489/7e7bb9da-d3f5-491b-bc5c-e1831ce66953">


The response after a CTRL+SHIFT+B then PVSneslib - Clean:
<img width="791" alt="Screenshot 2024-02-13 085812" src="https://github.com/alekmaul/pvsneslib/assets/25464489/28523b0e-d1d2-431a-8128-5f59a3bc8d2a">


I even removed the Windows Subsystem for Linux just to see if is was interfering.

**Reply by alekmaul** [reply]
it is really weird, I really don't understand where is the problem. I will talk about your problem on pvsneslib discord

**Comment by bkumanchik** [comment]
I've even tried another computer :(

**Comment by bkumanchik** [comment]
I want to feature it on here: 

https://www.youtube.com/playlist?list=PL6-0_fdvQrFPnfEJKpHxi6NPIADmreNdC

I've been doing this for over 3 years now and I'd love to show off pvsneslib but I'd like to show the whole pipeline. I really appreciate your patience, I can donate to the project if it would help.

Brian

**Reply by alekmaul** [reply]
donation will not help you but is appréciate. I think the issue is a msys issue evenif I really don't understand why. We will try to continue in the previous thread with your video and my last comments ;)

**Reply by bkumanchik** [reply]
Yes, maybe Unix/Linux syntax vs. MS DOS syntax formatting of command line stuff

**Reply by alekmaul** [reply]
I don't think so. as we are using unix like syntax, even with MS DOS. By the way, could you please reply to the questions on the previous thread regarding make and snes_rules?

**Reply by bkumanchik** [reply]
Make seems to be having trouble finding [snes_rules], is there a more direct way of including it? maybe a dumb question.

Also when I first open up the folder (from a fresh setup) I get this message, is that normal?

<img width="364" alt="Screenshot 2024-02-14 100943" src="https://github.com/alekmaul/pvsneslib/assets/25464489/157e51df-7800-4446-9239-6d3e7623d035">

**Reply by alekmaul** [reply]
i don't know if it is normal, sorry. I can't remember if I got this message the first time :(

**Comment by bkumanchik** [comment]
Thanks,
Not sure either, I have completely re-installed it more than a dozen times always trying something new and I just keep getting the same thing 
I appreciate the help, I will keep looking into the issue myself.
Brian    On Tuesday, February 13, 2024 at 11:52:41 PM EST, alekmaul ***@***.***> wrote:  
 
 


it is really weird, I really don't understand where is the problem. I will talk about your problem on pvsneslib discord

**Reply by alekmaul** [reply]
i can't remember: did you tried the make -v in a vscode terminal windows of hello_world example?

**Reply by bkumanchik** [reply]
make -v gave me this:

<img width="511" alt="Screenshot 2024-02-14 212807" src="https://github.com/alekmaul/pvsneslib/assets/25464489/10306025-01bf-48d8-972f-0e46d4d321f0">

**Reply by alekmaul** [reply]
my version is older than your and it works fine, i'm really out of tips about your problem :(
in this terminal windows (indside hello world example), it is still not working when you launch 'make clean' and then 'make'?

**Reply by alekmaul** [reply]
I think the best way to help you is to continue on discord, to see if someone else is blocked like you

**Comment by bkumanchik** [comment]
OK, thanks for all of your help.
Brian
    On Wednesday, February 14, 2024 at 11:35:36 PM EST, alekmaul ***@***.***> wrote:  
 
 


I think the best way to help you is to continue on discord, to see if someone else is blocked like you
